### PR TITLE
fix: clean error when no sites after site-filter

### PR DESCRIPTION
**Pull request by Teufelchen1**
Hi! 👋 

When accidentally running `website-stalker run all`, I got this error:
```
thread 'main' panicked at 'Site filter filtered everything out.', src/main.rs:130:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

which I disliked.

This PR changes that, the error now looks like this:
```
Error: The Site-filter filtered everything out.
There are no sites tracked for your query.
Tipp: Run 'website-stalker run --all' to stalk all tracked sites.
```

I believe this is better because:
- We can do a graceful exit here, no danger, no debugging needed, no panic
- Shows clearly this is an error - as opposed to failure/fault
- Shows what the user might did wrong
- Shows a possible solution for this error

Have a nice day!

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -137,7 +137,12 @@ async fn run(do_commit: bool, site_filter: Option<&Regex>) -> anyhow::Result<()>
         .filter(|site| site_filter.map_or(true, |filter| filter.is_match(site.url.as_str())))
         .collect::<Vec<_>>();
     let sites_amount = sites.len();
-    assert!(!sites.is_empty(), "Site filter filtered everything out.");
+    if sites.is_empty() {
+        eprintln!("Error: The Site-filter filtered everything out.
```
Comment (EdJoPaTo) [inline src/main.rs:141]:
```suggestion
        eprintln!("Error: The site-filter filtered everything out.
```

File: src/main.rs [hunk 2]
```diff
@@ -137,7 +137,12 @@ async fn run(do_commit: bool, site_filter: Option<&Regex>) -> anyhow::Result<()>
...
+There are no sites tracked for your query.
+Tipp: Run 'website-stalker run --all' to stalk all tracked sites.");
```
Comment (EdJoPaTo) [inline src/main.rs:143]: The current message is oddly specific to your `run all` case and should also include the option of another filter

```suggestion
Hint: Change the filter or use all sites with 'run --all'.");
```

File: src/main.rs [hunk 3]
```diff
@@ -137,7 +137,12 @@ async fn run(do_commit: bool, site_filter: Option<&Regex>) -> anyhow::Result<()>
         .filter(|site| site_filter.map_or(true, |filter| filter.is_match(site.url.as_str())))
         .collect::<Vec<_>>();
     let sites_amount = sites.len();
-    assert!(!sites.is_empty(), "Site filter filtered everything out.");
+    if sites.is_empty() {
+        eprintln!("Error: The Site-filter filtered everything out.
+There are no sites tracked for your query.
```
Comment (EdJoPaTo) [inline src/main.rs:142]: Not sure I like this sentence. Both 'tracked' and 'query' aren't used anywhere else currently. I think the other two sentences already explain what happened so maybe just omit this one?

**Review by EdJoPaTo — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 3 above)
Comment (Teufelchen1) [reply src/main.rs]: Hm. Fine with me.

**Review by EdJoPaTo — APPROVED** [review]

**Pull request merged**